theatre/studio/src/propEditors/useEditingToolsForCompoundProp.tsx

```diff
@@ -244,128 +250,114 @@ function ControlIndicators({
     we _could_ be re-using the calculation of `transform` in `position`, I think
     it's unlikely that this optimization would matter.
     */
-    const nearbyKeyframesInEachTrack = listOfDescendantTrackIds
-      .map((trackId) => ({
-        trackId,
-        track: val(
-          obj.template.project.pointers.historic.sheetsById[obj.address.sheetId]
-            .sequence.tracksByObject[obj.address.objectKey].trackData[trackId],
-        ),
-      }))
-      .filter(({track}) => !!track)
-      .map((s) => ({
-        ...s,
-        nearbies: getNearbyKeyframesOfTrack(
+    const nearbyKfsD = prism(function collectNearbyKeyframesForCompound() {
+      let hasCur = false
+      let closestPrev: undefined | KeyframeWithTrack = undefined
+      let closestNext: undefined | KeyframeWithTrack = undefined
+      let allCur = listOfDescendantTrackIds.length !== 0
+      // previously written as a series of functional methods, using a for loop is far faster, here.
+      // pipelined in an effort to keep the playhead scrubbing fast.
```

Comment:
aside: I find the functional methods easier to read, makes me wish there was a lib that allowed these chained methods to be executed in a single loop under the hood.